boxsdk/session/box_session.py

```diff
@@ -134,7 +141,10 @@ def with_shared_link(self, shared_link, shared_link_password=None):
         """
         headers = self._default_headers.copy()
         headers.update(get_shared_link_header(shared_link, shared_link_password))
-        return self.__class__(self._oauth, self._network_layer, headers)
+        return self.__class__(self._oauth, self._network_layer, headers, self._default_network_request_kwargs.copy())
+
+    def with_default_network_request_kwargs(self, **kwargs):
```

Comment:
This should probably be `request_data` (or whatever we call it), not `**kwargs`. One, for consistency and clarity. Two, because using `**kwargs` prevents us from being able to add any keyword arguments to this method (because adding any keyword arguments, which might be used for someone's `request_data`, would break them).

Since this is the only way for objects to specify `request_kwargs`, I think it makes more sense for this to be called `with_network_request_kwargs` (drop the default).

I also feel like the default behavior should be to `update` any existing arguments, not replace them. Otherwise, it's sort of a last-call-wins, and the "default" arguments only have a meaning if there are no more calls to this method.